### DISCUSSION TITLE
Have anyone stuck with error "Failed to start Ranch...." with Azimutt in container?

### DISCUSSION THREAD

**Discussion by NameArtem** [azimuttapp/azimutt#258] [Q&A]
Hi!
I got an Error in every containers with Azimutt, if I build container the error raises too.

[error] Failed to start Ranch listener AzimuttWeb.Endpoint.HTTP in :ranch_tcp:listen([cacerts: :..., key: :..., cert: :..., ip: {0, 0, 0, 0, 0, 0, 0, 0}, port: 4000]) for reason :eafnosupport (address family not supported by protocol family)
[](url)

![S1](https://github.com/azimuttapp/azimutt/assets/18335444/e34752e8-320d-4007-a40b-b0ec36d0fe56)

**Comment by loicknuchel** [comment]
Hi @NameArtem,

I missed alerts so just saw your message now, sorry for the delay.
I never had this error but looks like Azimutt is trying to start in [IPV6 mode](https://github.com/azimuttapp/azimutt/blob/main/backend/config/runtime.exs#L67) but your system doesn't support it.
I [made a fix](https://github.com/azimuttapp/azimutt/commit/94c889719578e4e3a272e1351aad3f68c07b3189) to add `PHX_PROTOCOL` environment variable to allow starting on IPv4 protocol.
Can you try it?

Thanks,
Loïc
